### DISCUSSION TITLE
Field of view as root property

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre-style-spec#144] [General]
The field of view is the opening angle of the camera frustum. In photography, it would be related to the camera focal length. A small fov angle corresponds to a long focal length and a large fov to a short focal length.

The field of view has an impact on fill extrusions, terrain 3d, and the pitch of the ground plain. It changes a lot how the map looks like. You can get a feeling for it with the following demo:

https://github.com/wipfli/field-of-view-demo

If I understand correctly, we have no way of specifying field of view in a style.json document at the moment. We can only set it via `map.transform.fov` in MapLibre GL JS.

Should we add a style spec root property called `fov` to specify field of view?

**Comment by wipfli** [comment]
It could be make the field of view dependent on pitch. Here is a demo where the fov interpolates between two values: 

https://wipfli.github.io/field-of-view-demo/interpolate#16/40.747266/-73.985667/0/45

**Comment by HarelM** [comment]
I think it might be tricky to properly define this so that it would make sense - what does 12 means? what does 35 means?
For me it feels like I only change the pitch, but I'm sure this is not the case, so, while this might be a great feature, I'm not sure how one would use it properly...

**Reply by wipfli** [reply]
I agree that field of view is a property which has a subtle effect in particular when interpolation based on pitch is used. But this is exactly the magic, interpolation lets you go from something which looks like a normal 2d map to something which looks like a normal 3d map. 

The field of view is defined as the opening angle of the camera frustum:

http://learnwebgl.brown37.net/08_projections/projections_perspective.html
